extensions/assetLibrary/src/node/index.ts

```diff
@@ -0,0 +1,251 @@
+
+      // if no sources are in the config file, set the default list to our 1st party feed.
+      if (!packageSources) {
+        packageSources = [
+          `https://gist.githubusercontent.com/benbrown/b932bbbf8b7c1583bbfb0cc70f051c62/raw/botframework-composer-packages.json`,
+        ];
+        composer.store.write('sources', packageSources);
+      }
+
+      let combined = [];
+      for (let s = 0; s < packageSources.length; s++) {
+        const url = packageSources[s];
+        try {
+          const raw = await axios.get(url);
+          if (Array.isArray(raw.data)) {
+            combined = combined.concat(raw.data);
+          } else {
+            console.error('Received non-JSON response from ', url);
+          }
+        } catch (err) {
+          console.error('Could not load library from URL');
+          console.error(err);
+        }
+      }
+      res.json(combined);
+    },
+    getComponents: async function (req, res) {
+      const user = await composer.context.getUserFromRequest(req);
+      const projectId = req.params.projectId;
+      const currentProject = await composer.getProjectById(projectId, user);
+      const runtime = composer.getRuntimeByProject(currentProject);
+      const mergeErrors: string[] = [];
+
+      const captureErrors = (msg: string): void => {
+        console.error(msg);
+        mergeErrors.push(msg);
+      };
+
+      if (currentProject.settings?.runtime?.customRuntime && currentProject.settings?.runtime?.path) {
+        const manifestFile = runtime.identifyManifest(currentProject.settings?.runtime?.path);
+
+        const dryrun = new SchemaMerger(
+          [manifestFile],
+          '',
+          path.join(currentProject.dataDir, 'dialogs/imported'),
+          true, // copy only? true = dry run
+          false, // verbosity: true = verbose
+          console.log,
+          console.warn,
+          captureErrors
+        );
+        const dryRunMergeResults = await dryrun.merge();
+
+        if (dryRunMergeResults) {
+          res.json({
+            components: dryRunMergeResults.components.filter((c) => c.includesSchema || c.includesExports),
+          });
+        } else {
+          res.status(500).json({
+            message: 'Could not load component list:' + mergeErrors.join('\n'),
+          });
+        }
+      } else {
+        res.json({
+          components: [],
+        });
+      }
+    },
+    import: async function (req, res) {
+      const user = await composer.context.getUserFromRequest(req);
+      const projectId = req.params.projectId;
+      const currentProject = await composer.getProjectById(projectId, user);
+      const runtime = composer.getRuntimeByProject(currentProject);
+
+      // get URL or package name
+      const packageName = req.query.package || req.body.package;
+      const version = req.query.version || req.body.version;
+      const isUpdating = req.query.isUpdating || req.body.isUpdating || false;
```

Comment:
Is there a need to support both the query string and body?

Reply:
no I will remove this, it made for easier testing initially but no longer needed.